### DISCUSSION TITLE
PubSub support

### DISCUSSION THREAD

**Discussion by GalaxyGorilla** [FreeOpcUa/python-opcua#1251] [Q&A]
Hey, I'm using the client and server to do some research on OPC UA and want to checkout how pubsub is working. Do you have any support for pubsub? Looking into the examples I don't see anything but the docu looks like there is actually some support for this.

Maybe I overlooked an example or searched for the wrong keywords? Maybe there is some good example floating around on github?

EDIT: sorry I think I mixed things up in the docu with the subscription service (which has nothing to do with pubsub). So for me it looks like there is no support for PubSub.

**Comment by oroulet** [comment]
You mean subscription? Or the new udp based protocol? The udp stuff is not implemented. Patch welcome

-------- Original Message --------

**Comment by GalaxyGorilla** [comment]
I mean subscriptions, but in the sense of PubSub (part 14 of the specs). Don't know if that stuff is new ... but yea it is UDP based when you go brokerless. Or you go with AMQP or MQTT (both use iirc TCP) when a broker is involved.

**Comment by oroulet** [comment]
Part 14 is not implemented. Patch welcome

-------- Original Message --------

**Comment by mila-mg** [comment]
Hello! I am fairly new to python and definitely a beginner in terms of opcua client installation. However, I am trying to create a client that is subscribed to a server running machine system values and want the client to be notified whenever specific values change.

My first question is: is this possible with python opcua? From what I understood, this has not been implemented yet?

My second question would be: what exactly do you mean when saying: "EDIT: sorry I think I mixed things up in the docu with the subscription service (which has nothing to do with pubsub)"? What is the difference between subscription service and publish-subscribe set up?

I am grateful for every help!

**Comment by AndreasHeine** [comment]
@mila-mg 

For Subsciprions we provide examples:
https://github.com/FreeOpcUa/python-opcua/tree/master/examples
https://github.com/FreeOpcUa/python-opcua/blob/0d595e596c9922699f0b36abb79583d5763eeb9b/examples/client-example.py#L76

The difference between subscription and pub/sub is:

Subscription: https://documentation.unified-automation.com/uasdkhp/1.0.0/html/_l2_ua_subscription.html
Pub/Sub: UDP Broadcasting / AMQP / MQTT Protocols so its for fast realtimecommunication (in difference to the subscription which is a cyclic polling for queued valuechanges "publishrequest / publishresponse" mecanisme)
